CMakeLists.txt

```diff
@@ -186,6 +186,8 @@ option(BUILD_TESTING "Build the testing tree." OFF)
 # Disable building the examples by default until the Idlpp-cxx has been
 # deprecated.
 option(BUILD_EXAMPLES "Build examples." OFF)
+# Enable SHM by default
+option(ENABLE_SHM "Enable shared memory support" ON)
```

Comment:
I can set it to OFF by default, but if I set this to `ON` when I can get the property `SHM_SUPPORT_IS_AVAILABLE` then enabling shared memory is automatic and the user doesn't have any control on SHM for the C++ binding, since it will be enabled automatically if Cyclone is built with SHM support, is this the desired behavior?

If yes then we don't need to even provide this option `ENABLE_SHM`, correct?

Reply:
My whole reasoning behind the logic of `ENABLE_SHM` option and property `SHM_SUPPORT_IS_AVAILABLE` is as below:
1.  Cyclone DDS C++ binding has `ENABLE_SHM` option to enable SHM with C++ binding
1. If the `ENABLE_SHM` option is set to `ON` when compiling Cyclone DDS C++ binding, then we check for the property `SHM_SUPPORT_IS_AVAILABLE` from `CycloneDDS::ddsc` 
    1. If this property is available then we set `DDSCXX_HAS_SHM` variable, compile C++ binding with SHM support and generate a `features.hpp` header with `DDSCXX_HAS_SHM` defined to 1. Lastly also add shared memory tests to `ddscxx_tests`
    2. If this property is unavailable (which means Cyclone DDS is compiled without SHM support), then with a warning we just compile C++ binding without any SHM support or shared memory tests.
1. If the `ENABLE_SHM` option is set to `OFF` we compile the C++ binding as it was before (without any SHM support)

Let me know what is the expected behavior here and I can modify the logic accordingly